### PR TITLE
[WIP] flax gradient checkpointing

### PR DESCRIPTION
**Pull request by patil-suraj**
# What does this PR do?

Adds gradient checkpointing to flax models

### CODE REVIEW

File: src/transformers/models/gpt2/modeling_flax_gpt2.py [hunk 1]
```diff
@@ -425,8 +425,9 @@ class FlaxGPT2BlockCollection(nn.Module):
     dtype: jnp.dtype = jnp.float32
 
     def setup(self):
+        block_cls = nn.remat(FlaxGPT2Block, concrete=True) if self.config.gradient_checkpointing else FlaxGPT2Block
```
Comment (patrickvonplaten) [inline src/transformers/models/gpt2/modeling_flax_gpt2.py:428]: Let's actually not make it dependent on the config here. @sgugger made a good point a while back that it shouldn't have been in the config in the first place. Can we maybe just add an input parameter to the model class init that is then called from the training scripts?

**Comment by patrickvonplaten** [comment]
Would be good to get some results on how much memory can be saved and if this is correctly implemented

**Comment by patil-suraj** [comment]
unstale

**Comment by ArthurZucker** [comment]
Closing as this PR is super old
